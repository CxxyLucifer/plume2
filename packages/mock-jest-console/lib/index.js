"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var BaseMock = /** @class */ (function () {
    function BaseMock() {
        this._logs = [];
        this.mock();
    }
    BaseMock.prototype.mock = function () { };
    BaseMock.prototype.mockFn = function () {
        var _this = this;
        return jest.fn(function (log) { return _this._logs.push(log); });
    };
    Object.defineProperty(BaseMock.prototype, "logs", {
        get: function () {
            return this._logs;
        },
        enumerable: true,
        configurable: true
    });
    BaseMock.prototype.reset = function () {
        this._logs = [];
    };
    return BaseMock;
}());
exports.BaseMock = BaseMock;
var MockWarn = /** @class */ (function (_super) {
    __extends(MockWarn, _super);
    function MockWarn() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockWarn.prototype.mock = function () {
        console.warn = this.mockFn();
    };
    return MockWarn;
}(BaseMock));
exports.MockWarn = MockWarn;
var MockLog = /** @class */ (function (_super) {
    __extends(MockLog, _super);
    function MockLog() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockLog.prototype.mock = function () {
        console.log = this.mockFn();
    };
    return MockLog;
}(BaseMock));
exports.MockLog = MockLog;
var MockDir = /** @class */ (function (_super) {
    __extends(MockDir, _super);
    function MockDir() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockDir.prototype.mock = function () {
        console.dir = this.mockFn();
    };
    return MockDir;
}(BaseMock));
exports.MockDir = MockDir;
var MockDebug = /** @class */ (function (_super) {
    __extends(MockDebug, _super);
    function MockDebug() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockDebug.prototype.mock = function () {
        console.debug = this.mockFn();
    };
    return MockDebug;
}(BaseMock));
exports.MockDebug = MockDebug;
var MockError = /** @class */ (function (_super) {
    __extends(MockError, _super);
    function MockError() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockError.prototype.mock = function () {
        console.error = this.mockFn();
    };
    return MockError;
}(BaseMock));
exports.MockError = MockError;
var MockTrace = /** @class */ (function (_super) {
    __extends(MockTrace, _super);
    function MockTrace() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockTrace.prototype.mock = function () {
        console.trace = this.mockFn();
    };
    return MockTrace;
}(BaseMock));
exports.MockTrace = MockTrace;
var MockConsole = /** @class */ (function (_super) {
    __extends(MockConsole, _super);
    function MockConsole() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockConsole.prototype.mock = function () {
        console.warn = this.mockFn();
        console.log = this.mockFn();
        console.dir = this.mockFn();
        console.debug = this.mockFn();
        console.error = this.mockFn();
        console.trace = this.mockFn();
    };
    return MockConsole;
}(BaseMock));
exports.MockConsole = MockConsole;
