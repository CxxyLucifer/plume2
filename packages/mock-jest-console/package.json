{
  "name": "mock-jest-console",
  "version": "1.1.1",
  "description": "jest.fn mock console.xxx",
  "main": "./lib/index.js",
  "types": "./lib/typings/index.d.ts",
  "keywords": [
    "mock jest console"
  ],
  "author": "hufeng",
  "license": "MIT"
}
