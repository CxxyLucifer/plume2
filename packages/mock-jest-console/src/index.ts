declare const jest: { fn: Function };

export class BaseMock {
  constructor() {
    this._logs = [];
    this.mock();
  }

  private _logs;

  mock() {}

  mockFn() {
    return jest.fn(log => this._logs.push(log));
  }

  get logs() {
    return this._logs;
  }

  reset() {
    this._logs = [];
  }
}

export class MockWarn extends BaseMock {
  mock() {
    console.warn = this.mockFn();
  }
}

export class MockLog extends BaseMock {
  mock() {
    console.log = this.mockFn();
  }
}

export class MockDir extends BaseMock {
  mock() {
    console.dir = this.mockFn();
  }
}

export class MockDebug extends BaseMock {
  mock() {
    console.debug = this.mockFn();
  }
}

export class MockError extends BaseMock {
  mock() {
    console.error = this.mockFn();
  }
}

export class MockTrace extends BaseMock {
  mock() {
    console.trace = this.mockFn();
  }
}

export class MockGroup extends BaseMock {
  mock() {
    console.group = this.mockFn();
  }
}

export class MockGroupCollapsed extends BaseMock {
  mock() {
    console.groupCollapsed = this.mockFn();
  }
}

export class MockTime extends BaseMock {
  mock() {
    console.time = this.mockFn();
  }
}

export class MockConsole extends BaseMock {
  mock() {
    console.warn = this.mockFn();
    console.log = this.mockFn();
    console.dir = this.mockFn();
    console.debug = this.mockFn();
    console.error = this.mockFn();
    console.trace = this.mockFn();
    console.group = this.mockFn();
    console.groupCollapsed = this.mockFn();
    console.time = this.mockFn();
  }
}
