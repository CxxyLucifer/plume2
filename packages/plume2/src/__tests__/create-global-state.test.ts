import { fromJS } from 'immutable';
import { MockConsole } from 'mock-jest-console';
import { Action, Actor, createGlobalState, IMap, QL, RL, Store } from '..';

jest.mock('raf', () => jest.fn(cb => cb()));

const { getter, setter } = createGlobalState({
  hello: 'hello global state'
});

const helloQL = QL('helloQL', ['hello', hello => hello]);
const helloRL = RL('helloRL', [
  helloQL,
  hello => {
    setter({ hello });
  }
]);

class HelloActor extends Actor {
  defaultState() {
    return { hello: 'hello' };
  }

  @Action('globalState')
  globalStateChange(state: IMap, param) {
    return state.merge(param);
  }

  @Action('change')
  ChannelMergerNode(state: IMap, param) {
    return state.set('hello', param);
  }
}

class ShoppingCartStore extends Store {
  bindActor() {
    return [HelloActor];
  }

  bindRx() {
    return {
      helloRL
    };
  }

  bindGlobalState() {
    getter(state => {
      this.dispatch('globalState', state);
    });
  }
}

class ProductListStore extends Store {
  bindActor() {
    return [HelloActor];
  }

  bindGlobalState() {
    getter(state => this.dispatch('globalState', state));
  }
}

it('test merge global state', () => {
  const store = new ShoppingCartStore();
  const state = store.state();
  expect(state.toJS()).toEqual({ hello: 'hello global state' });
});

it('test setGlobalState callback', () => {
  const store = new ShoppingCartStore();
  setter(state => state.set('hello', 'I was changed by set global state'));
  expect(store.state().toJS()).toEqual({
    hello: 'I was changed by set global state'
  });
});

it('test setGlobalState value', () => {
  const store = new ShoppingCartStore();
  setter(
    fromJS({
      hello: 'I was changed by set global state'
    })
  );
  expect(store.state().toJS()).toEqual({
    hello: 'I was changed by set global state'
  });
});

it('multiple store', () => {
  //reset
  setter(fromJS({ hello: 'hello global state' }));

  const shoppinCartStore = new ShoppingCartStore();
  const productListStore = new ProductListStore();

  expect(shoppinCartStore.state().toJS()).toEqual({
    hello: 'hello global state'
  });
  expect(productListStore.state().toJS()).toEqual({
    hello: 'hello global state'
  });
});

it('multiple store complex secene', () => {
  const mock = new MockConsole();
  setter(fromJS({ hello: 'hello global state' }));
  const shoppinCartStore = new ShoppingCartStore({ debug: true });
  const productListStore = new ProductListStore({ debug: true });
  expect(shoppinCartStore.state().toJS()).toEqual({
    hello: 'hello global state'
  });
  expect(productListStore.state().toJS()).toEqual({
    hello: 'hello global state'
  });

  shoppinCartStore.dispatch('change', 'multiple store complex secene');

  expect(shoppinCartStore.state().toJS()).toEqual({
    hello: 'multiple store complex secene'
  });
  expect(productListStore.state().toJS()).toEqual({
    hello: 'multiple store complex secene'
  });

  expect(mock.logs).toMatchSnapshot();
});
