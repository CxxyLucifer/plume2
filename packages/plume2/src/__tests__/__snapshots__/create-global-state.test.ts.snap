// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple store complex secene 1`] = `
Array [
  "Store init ðŸš€",
  "Actor:reduce HelloActor",
  "parse:rl -> helloRL",
  "ðŸ”¥:tracing: QL(helloQL)",
  "dep:hello, cache:false, value:\\"hello\\"",
  "QL(helloQL)|> \\"hello\\"",
  "store dispatch => 'globalState'",
  "params|> immutable[yes] ",
  Object {
    "hello": "hello global state",
  },
  "HelloActor receive => 'globalState'",
  "Reactive ðŸš€",
  "ðŸ”¥:tracing: QL(helloQL)",
  "dep:hello, cache:false, value:\\"hello global state\\"",
  "QL(helloQL)|> \\"hello global state\\"",
  "helloRL => ðŸ”¥",
  "Store init ðŸš€",
  "Actor:reduce HelloActor",
  "store dispatch => 'globalState'",
  "params|> immutable[yes] ",
  Object {
    "hello": "hello global state",
  },
  "HelloActor receive => 'globalState'",
  "store dispatch => 'change'",
  "params|> multiple store complex secene",
  "HelloActor receive => 'change'",
  "Reactive ðŸš€",
  "ðŸ”¥:tracing: QL(helloQL)",
  "dep:hello, cache:false, value:\\"multiple store complex secene\\"",
  "QL(helloQL)|> \\"multiple store complex secene\\"",
  "helloRL => ðŸ”¥",
  "store dispatch => 'globalState'",
  "params|> immutable[yes] ",
  Object {
    "hello": "multiple store complex secene",
  },
  "HelloActor receive => 'globalState'",
  "store dispatch => 'globalState'",
  "params|> immutable[yes] ",
  Object {
    "hello": "multiple store complex secene",
  },
  "HelloActor receive => 'globalState'",
]
`;
